wormchain/contracts/tools/test_wormchain.ts

```diff
@@ -0,0 +1,180 @@
+
+  const nativeBalance = await cwc.getBalance(signer, denom)
+  console.log("nativeBalance ", nativeBalance.amount)
+
+  const utestBalance = await cwc.getBalance(signer, "utest")
+  console.log("utest balance ", utestBalance.amount)
+
+
+  // create key for guardian0
+  const ec = new elliptic.ec("secp256k1");
+  // create key from the devnet guardian0's private key
+  const key = ec.keyFromPrivate(Buffer.from(signerPk, "hex"));
+
+
+  // Test empty observation
+
+  // object to json string, then to base64 (serde binary)
+  const arrayBinaryString = toBinary([]);
+
+  // combine parts to be Uint8Array with length 65
+  const signedEmptyArray = signBinary(key, arrayBinaryString)
+
+  const observeEmptyArray = {
+    submit_observations: {
+      observations: arrayBinaryString,
+      guardian_set_index: 0,
+      signature: {
+        index: 0,
+        signature: Array.from(signedEmptyArray),
+      },
+    },
+  };
+
+  let emptyArrayObsRes = await cwc.execute(signer, accountingAddress, observeEmptyArray, "auto");
+  console.log(`emptyArrayObsRes.transactionHash: ${emptyArrayObsRes.transactionHash}`);
+
+
+  // Test (fake) observation
+  const emitter_address = "0000000000000000000000000290fb167208af455bb137780163b7b7a9a10c16"
+  const observations = [
+    {
+      emitter_chain: 2,
+      emitter_address: emitter_address,
+      sequence: 2,
+      nonce: 1,
+      consistency_level: 0,
+      timestamp: 1,
+      payload:
+        Buffer.from("030000000000000000000000000000000000000000000000000000000005f5e1000000000000000000000000002d8be6bf0baa74e0a907016679cae9190e80dd0a0002000000000000000000000000c10820983f33456ce7beb3a046f5a83fa34f027d0c2000000000000000000000000000000000000000000000000000000000000f4240", "hex").toString("base64"),
+
+      tx_hash:
+        Buffer.from("9fc68fb0ee735d45c9074a20adef1747b0593803f33b9f3f2252c8e2df567f41", "hex").toString("base64")
+    },
+  ];
+
+  // object to json string, then to base64 (serde binary)
+  const observationsBinaryString = toBinary(observations);
+
+  const signed = signBinary(key, observationsBinaryString)
+
+  const executeMsg = {
+    submit_observations: {
+      observations: observationsBinaryString,
+      guardian_set_index: 0,
+      signature: {
+        index: 0,
+        signature: Array.from(signed),
+      },
+    },
+  };
+  console.log(executeMsg);
+
+  let inst = await cwc.execute(
+    signer,
+    accountingAddress,
+    executeMsg,
+    "auto"
+  );
+  let txHash = inst.transactionHash;
+  console.log(`executed submit_observation! txHash: ${txHash}`);
+
+
+  // send it again and expect it to fail
+  try {
+    await cwc.execute(
+      signer,
+      accountingAddress,
+      executeMsg,
+      "auto"
+    );
+    throw new Error("submitting observation twice succeeded.")
+  } catch (_) {
+    // expected it to fail with:
+    // "failed to handle `Observation`: message already processed"
+    console.log("sending duplicate observations fails as expected.")
```

Comment:
This will no longer be an error once #2241 merges.  Instead we'll return a response with an error for the observation status but the tx itself will succeed.

Reply:
Good to know, thanks for the flag.

Reply:
removing